lib/ecco/lifecycle_failure_listener.rb

```diff
@@ -0,0 +1,14 @@
+module Ecco
+  class LifecycleFailureListener < com.github.shyiko.mysql.binlog.BinaryLogClient::AbstractLifecycleListener
+    attr_writer :callback
+
+    def initialize
+      @callback = Proc.new {}
+    end
+
+    def onCommunicationFailure(client, exception)
```

Comment:
Just a note here, it was not possible to name this `on_communication_failure` since we override a method in the Abstract base class `AbstractLifecycleListener`. JRuby did not "understand" what to override.

With interfaces JRuby does understand which method to implement though.